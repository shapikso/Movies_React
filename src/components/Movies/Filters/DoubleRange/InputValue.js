import React from 'react';

const InputValue = ({value}) => (<input type="text" className="double-range__info-min basic-field" value={value}/>);

export default InputValue;