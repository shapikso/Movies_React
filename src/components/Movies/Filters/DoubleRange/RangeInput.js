import React from 'react';

const RangeInput = ({minValue,maxValue, step, value}) =>  (
    <input  type="range" className="double-range__place-item" min={minValue} max={maxValue} step={step} value={value} />);

export default RangeInput;