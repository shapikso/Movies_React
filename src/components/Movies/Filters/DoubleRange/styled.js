import styled from "styled-components";

export const StDoubleRange = styled.div`
  margin-bottom: 40px;
  border: 1px solid var(--light-primary-color);
  border-radius: 5px;
  padding: 14px;
  margin-bottom: 15px;
    `;

export const StDoubleRangeTittle = styled.div`
  margin-bottom: 15px;
    `;
export const StDoubleRangeInfo = styled.div`
  width: 250px;
  display: flex;
  justify-content: space-between;
  align-items: center;
    `;
export const StDoubleRangeInfoMinMax = styled.div`
  width: 97%;
  background: none;
  color: var(--light-primary-color);
  padding: 10px;
  text-align: center;
    `;

export const StDoubleRangePlace= styled.div`
  width: 100%;
  position: relative;
  height: 20px;
  cursor: pointer;
    `;

export const StDate= styled.div`
  width: 48%;
  position: relative;
    `;
export const StSelect = styled.select`
  padding: 10px 25px;
  background: none;
  cursor: pointer;
  border: 1px solid var(--light-primary-color);
  border-radius: 5px;
  height: 50px;
  font-size: 16px;
  color: var(--light-primary-color);
    `;
