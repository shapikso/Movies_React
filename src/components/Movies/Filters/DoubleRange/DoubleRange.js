import React from 'react';
import InputField from "../../../common/Input/InputField";
import PropTypes from 'prop-types';
import Slider from 'rc-slider';
import './slider.scss';
import {StDoubleRange, StDoubleRangeInfo, StDoubleRangePlace, StDoubleRangeTittle} from "./styled";
const { createSliderWithTooltip } = Slider;
const Range = createSliderWithTooltip(Slider.Range);

const DoubleRange = ({title,minValue,maxValue, step, setRange, minInputValue, maxInputValue}) => {

    const onChangeRangeHandler = ([key,value]) => setRange(key,value);

    return (
        <StDoubleRange>
            <StDoubleRangeTittle>{title}</StDoubleRangeTittle>
            <StDoubleRangeInfo>
                <InputField className="double-range__info-min" readOnly={true} value={minInputValue} />
                <InputField className="double-range__info-min" readOnly={true} value={maxInputValue} />
            </StDoubleRangeInfo>
            <StDoubleRangePlace>
                <Range
                    min={+minValue}
                    max={+maxValue}
                    step={step}
                    defaultValue={[+minValue, +maxValue]}
                    onChange={onChangeRangeHandler}
                />
            </StDoubleRangePlace>
        </StDoubleRange>
    );
};

DoubleRange.propTypes = {
    minValue: PropTypes.string.isRequired,
    maxValue: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired,
    step: PropTypes.string.isRequired,
    maxInputValue: PropTypes.number.isRequired,
    minInputValue: PropTypes.number.isRequired,
    setRange : PropTypes.func.isRequired
};

export default DoubleRange;