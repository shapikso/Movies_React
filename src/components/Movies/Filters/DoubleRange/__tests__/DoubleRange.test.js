import React from 'react';
import {shallow, mount} from 'enzyme';
import DoubleRange from "../DoubleRange";

describe('DoubleRange', ()=> {
    const props = {
        title: 'Base title',
        minValue: '0',
        maxValue: '400',
        step: '2',
        setRange: jest.fn(),
        minInputValue: 0,
        maxInputValue: 400
    };
    it('should render correctly', () => {
        const component = shallow(<DoubleRange {...props} />);
        expect(component).toMatchSnapshot();
    });
    it('should checked render title', () => {
        const component = mount(<DoubleRange {...props} />);
        expect(component.find('div div').at(0).text()).toEqual(props.title);
    });
    it('should checked render min value', () => {
        const component = shallow(<DoubleRange {...props} />);
        expect(component.find('InputField').at(0).getElement().props.value).toEqual(props.minInputValue);
    });
    it('should checked render max value', () => {
        const component = shallow(<DoubleRange {...props} />);
        expect(component.find('InputField').at(1).getElement().props.value).toEqual(props.maxInputValue);
    });
    it('should checked render min value range', () => {
        const component = shallow(<DoubleRange {...props} />);
        expect(component.find('ComponentWrapper').getElement().props.min).toEqual(+props.minValue);
    });
    it('should checked render max value range', () => {
        const component = shallow(<DoubleRange {...props} />);
        expect(component.find('ComponentWrapper').getElement().props.max).toEqual(+props.maxValue);
    });
    it('should change range', () => {
        const component = shallow(<DoubleRange {...props} />);
        component.find('ComponentWrapper').getElement().props.onChange('59');
        expect(props.setRange).toHaveBeenCalledWith('5','9');
    });
});