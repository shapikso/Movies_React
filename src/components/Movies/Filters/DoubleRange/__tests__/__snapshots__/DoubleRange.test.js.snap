// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DoubleRange should render correctly 1`] = `ShallowWrapper {}`;
